### ISSUE TITLE
cpu_temp on rasberry PI 4

### ISSUE THREAD

**Issue by GeyerA** [matthewwall/weewx-cmon#2]
Hi,
I am using your extension for a while and tried to get the cpu_temp to work. I also tried to use the core_temp but still did not succeed. I run a raspberry PI 4 on Archlinux. Any hints to get it to work? Thanks

**Comment by bobclare** [comment]
I had the same issue on my rpi 3.  I discovered that the check in the temperature code for the "special" location of the information on the raspberry pi is inverted.  The problem is that there is a check on "tdir" followed by a check on "tfile".  Unfortunately, at least on my rpi, the directory tdir exists, but just isn't populated with the right info.  The check that is done is:
if exists tdir then get temp info
elif exists tfile then get temp info

I just swapped the order of the checks and it works for me.  I don't have access to any other linux based machine, so I can't tell whether tfile exists on generic linux machines to know whether the swapped test would work on other linux-based machines.

**Comment by GeyerA** [comment]
Hi, thanks for the hint.
I tried but still does not work. The interesting part is that both paths should work. I got the CPU temp in the "tdir" and "tfile". Did you also enable the "measure rpi attributes (not all platforms support this)". I altered and ran it with all possible combiations, still did not return anything.

**Comment by bobclare** [comment]
Hmmm...  I did not enable the extra rpi attributes.  As far as I could tell, doing so would change the schema and I would probably lose all the archived data that I had (not sure about that though).  But definitely, for my rpi, there's no temp data in the tdir, but only in tfile and changing the order of checking gives me cpu temps now that I never had before.

**Comment by GeyerA** [comment]
Exactly, I even deleted the cmon.sdb since it could not resolve the new structure after re-starting weewx.
My cmon.py looks now like this.
`# read cpu temperature

        tdir = '/sys/class/hwmon/hwmon0/device'
        # rpi keeps cpu temperature in a different location

        tfile = '/sys/class/thermal/thermal_zone0/temp'
        if os.path.exists(tfile):

            try:

                s = self._readproc_line(tfile)

                t_C = int(s) / 1000 # degree C

                record['cpu_temp'] = t_C

            except Exception as e:

                logdbg("read failed for %s: %s" % (tfile, e))

        elif os.path.exists(tdir):

            try:

                for f in os.listdir(tdir):

                    if f.endswith('_input'):

                        s = self._readproc_line(os.path.join(tdir, f))

                        if s and len(s):

                            n = f.replace('_input', '')

                            t_C = int(s) / 1000 # degree C

                            record['cpu_' + n] = t_C

            except Exception as e:

                logdbg("read failed for %s: %s" % (tdir, e))
`
